### PR TITLE
Feature spawncore conf

### PR DESCRIPTION
**Pull request by vEpiphyte**
Allow the spawncore to be a configurable object

### CODE REVIEW

File: synapse/cortex.py [hunk 1]
```diff
@@ -890,6 +890,7 @@ async def finidmon():
         import synapse.lib.spawn as s_spawn  # get around circular dependency
         self.spawnpool = await s_spawn.SpawnPool.anit(self)
         self.onfini(self.spawnpool)
+        self.spawncorector = None
```
Comment (vEpiphyte) [inline synapse/cortex.py:893]: It's that we can store the ctor here `self.spawncorector = f'{s_spawn.__name__}.{s_spawn.SpawnCore.__name__}'`
but that make it a bit ickier to subclass the Cortex.

```class SubCortex(s_cortex.Cortex):

    async def __anit__(self, dirn, conf=None):
        await s_cortex.Cortex.__anit__(self, dirn, conf=conf)
        self.spawncorector = f'{s_some_spawn.__name__}.{s_some_spawn.SomeSpawnCore.__name__}'
```

feels icky

File: synapse/cortex.py [hunk 2]
```diff
@@ -1007,7 +1008,24 @@ async def getSpawnInfo(self):
                 'pkgs': await self.getStormPkgs(),
             },
             'model': await self.getModelDefs(),
+            'spawncorector': await self.getSpawncoreCtor()
         }
+        addl = await self._addlSpawnInfo()
+        ret.update(**addl)
+        return ret
+
+    async def _addlSpawnInfo(self):
```
Comment (invisig0th) [inline synapse/cortex.py:1017]: This should be facilitated in a subclass by overriding getSpawnInfo

File: synapse/cortex.py [hunk 3]
```diff
@@ -1007,7 +1008,24 @@ async def getSpawnInfo(self):
...
+        '''
+        Additional spawninfo data.
+        '''
+        return {}
+
+    async def getSpawncoreCtor(self):
```
Comment (invisig0th) [inline synapse/cortex.py:1023]: This should just implement return s_spawn and allow any subclass to override

File: synapse/cortex.py [hunk 4]
```diff
@@ -1007,7 +1008,24 @@ async def getSpawnInfo(self):
...
+        if self.spawncorector:
+            return self.spawncorector
+        import synapse.lib.spawn as s_spawn
```
Comment (invisig0th) [inline synapse/cortex.py:1026]: import within a function?

File: synapse/cortex.py [hunk 5]
```diff
@@ -1007,7 +1008,24 @@ async def getSpawnInfo(self):
...
+        self.spawncorector = f'{s_spawn.__name__}.{s_spawn.SpawnCore.__name__}'
```
Comment (invisig0th) [inline synapse/cortex.py:1027]: This is no less hard-coded than simply returning "synapse.lib.spawn.SpawnCore".  Also i'm not a fan of having an accessor method *and* a local value. :)

File: synapse/cortex.py (hunk 1 above)
Comment (invisig0th) [reply synapse/cortex.py:893]: See my comment below ( we probably shouldn't store it at all )

File: synapse/cortex.py (hunk 4 above)
Comment (vEpiphyte) [reply synapse/cortex.py:1026]: we do import inside of `__anit__` already.

**Pull request merged**